src/providers/wfs/qgswfsdataitems.cpp

```diff
@@ -50,6 +55,76 @@ QgsWfsLayerItem::~QgsWfsLayerItem()
 {
 }
 
+QList<QMenu *> QgsWfsLayerItem::menus()
+{
+  QList<QMenu *> menus;
+
+  if ( mPath.startsWith( QLatin1String( "geonode:/" ) ) )
+  {
+    QAction *actionCopyStyle = new QAction( tr( "Copy Style" ), this );
+    connect( actionCopyStyle, &QAction::triggered, this, &QgsWfsLayerItem::copyStyle );
+
+    QMenu *menuStyleManager = new QMenu();
```

Comment:
menuStyleManager needs to have its parent set to `this`, or it will be leaked

Reply:
I'm not sure this belongs here. All provider data item implementations use the inheritance patter to implement specialized behaviors. I'd suggest you to follow the same pattern, unless there is a very good reason for not doing so.

Reply:
If I read the docs, QMenu parent is a QWidget even though the QgsDataItem isn't a QWidget.

Reply:
Ah - you're right. In that case you'll need to return ownership of the menus when this is called, and reparent them in the caller.

Reply:
Or add a `QWidget *parent` parameter to the method, I guess that would be understandable for devs even without reading the docs.